instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -218,6 +230,32 @@ def _operation_name(self, db_name, statement):
             return self.vendor
         return " ".join(parts)
 
+    def _get_commenter_data(self, conn) -> dict:
+        """Calculate sqlcomment contents from conn and configured options"""
+        commenter_data = {
+            "db_driver": conn.engine.driver,
+            # Driver/framework centric information.
+            "db_framework": f"sqlalchemy:{sqlalchemy.__version__}",
+        }
+
+        if self.commenter_options.get("opentelemetry_values", True):
+            commenter_data.update(**_get_opentelemetry_values())
+
+        # Filter down to just the requested attributes.
+        commenter_data = {
+            k: v
+            for k, v in commenter_data.items()
+            if self.commenter_options.get(k, True)
+        }
+        return commenter_data
+
+    def _set_db_client_span_attributes(self, span, statement, attrs) -> None:
+        """Uses statement and attrs to set attributes of provided Otel span"""
+        span.set_attribute(SpanAttributes.DB_STATEMENT, statement)
```

Comment:
Prob isn't related, but shouldn't these attributes be provided at span creation?  For sampling decisions.

Reply:
If so, maybe we can pass them at span creation and convert this method to a staticmethod that just update the statement with the statement with commenter_data.

Reply:
If so, maybe we can pass them at span creation and convert this method to a staticmethod that just update the statement with the statement with commenter_data.

Reply:
The calls to `set_attribute` in this instrumentor generally have been set to only happen if `span.is_recording()`, which for this (new) helper is at [L272](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3112/files#diff-4bf18e7d840238995e3eb1152d6bf0a3f56ce6ab6f2bcf1726ecbac7a79afedbR272). This condition was introduced in https://github.com/open-telemetry/opentelemetry-python/issues/1057 to improve performance, so I think that part should be kept in. Then regarding staticmethod, I'm not sure if it'll be possible but maybe I miss something. Thoughts? 🙂